go/vt/tabletserver/planbuilder/dml.go

```diff
@@ -29,6 +29,11 @@ func analyzeUpdate(upd *sqlparser.Update, getTable TableGetter) (plan *ExecPlan,
 		return nil, err
 	}
 
+	// Store the WHERE clause as string for the transaction serializer.
+	buf := sqlparser.NewTrackedBuffer(nil)
+	buf.Myprintf("%v", upd.Where)
```

Comment:
On second thoughts, I don't think this is safe. If you don't include the table name in WhereClause, you'll start unnecessarily serializing all statements that have the same where clauses, like "where id = 1", which may be against different tables.
You can safely assume that there will always be a table name for DMLs. So, it's better to add it to the string to match.